plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -193,5 +197,18 @@ private static void generateEnumAttributeSchema(JSONObject attributeSchemaTempla
                     .put("enum", new JSONArray(attributeList)));
         }
     }
+
+    public static void rootConfigGeneration() throws Exception {
+        DefaultConfiguratorRegistry registry = new DefaultConfiguratorRegistry();
+        final ConfigurationContext context = new ConfigurationContext(registry);
+        for (RootElementConfigurator root : RootElementConfigurator.all()) {
+            final CNode config = root.describeStructure(root.getTargetComponent(context), context);
+            final Mapping mapping = config.asMapping();
+            final List<Map.Entry<String, CNode>> entries = new ArrayList<>(mapping.entrySet());
+            for (Map.Entry<String, CNode> entry : entries) {
+                System.out.println(entry.getKey() + " " + entry.getValue().toString());
```

Comment:
@casz I cannot quite iterate over the `entry.getValue()` CNode. Because it returns the Root configs but I still need to get out of it the further attribtutes. ?

Reply:
Also Some attributes still return NUll

Reply:
can you put ideas here please so don't need to jump between gitter and github, or at least copy the conversation in

Reply:
Yeah sounds like I should put in the ideas here, basically need to iterate over `entry.getValue()` since it is a CNode. By Iterate I mean just go over the entire structure present inside the CNode